### PR TITLE
Update WorkToDo with Config from LE

### PR DESCRIPTION
**Pull request by surags**
https://rally1.rallydev.com/#/detail/userstory/422999259560?fdp=true

### CODE REVIEW

File: components/d2l-work-to-do/d2l-work-to-do.js [hunk 1]
```diff
@@ -130,6 +134,23 @@ class WorkToDoWidget extends EntityMixinLit(LocalizeMixin(LitElement)) {
...
+					break;
+			}
+		});
+	}
+
+	_optionsChanged(newOptions) {
+		const options = JSON.parse(newOptions);
+		this.fullscreen = options.fullscreen;
+		this._overdueWeekLimit = options.overdueWeekLimit;
+		this._upcomingWeekLimit = options.upcomingWeekLimit;
```
Comment (KearseTrevor) [inline components/d2l-work-to-do/d2l-work-to-do.js:150]: In the case that options doesn't carry these properties, do the values for the variables stay at what is set in the constructor? Or do they get reassigned to undefined? If the later, we'll require || assignments

File: components/d2l-work-to-do/d2l-work-to-do.js [hunk 2]
```diff
@@ -130,6 +134,23 @@ class WorkToDoWidget extends EntityMixinLit(LocalizeMixin(LitElement)) {
...
+		window.D2L.options = options;
```
Comment (KearseTrevor) [inline components/d2l-work-to-do/d2l-work-to-do.js:151]: Are we positive we aren't stepping on anyone's toes by using this window property? Potentially may want something more specific to WorkToDo (ex: window.D2L.WorkToDoOptions)

Comment (surags) [reply components/d2l-work-to-do/d2l-work-to-do.js]: Renamed

File: components/d2l-work-to-do/d2l-work-to-do.js (hunk 1 above)
Comment (surags) [reply components/d2l-work-to-do/d2l-work-to-do.js]: Now only updates if exists in the options object

**Comment by KearseTrevor** [comment]
In holding pattern until conversation in https://github.com/Brightspace/lms/pull/4295 (sibling PR on LE side) is put further along.  Suggest potentially moving this PR back to draft form if it looks like things are going to get shook up a reasonable amount over there.

File: components/d2l-work-to-do/d2l-work-to-do.js [hunk 3]
```diff
@@ -139,6 +147,29 @@ class WorkToDoWidget extends EntityMixinLit(LocalizeMixin(LitElement)) {
...
 	}
 
+	attributeChangedCallback(name, oldval, newval) {
+		if (!window.D2L.workToDoOptions) {
+			window.D2L.workToDoOptions = {};
+		}
+
+		switch (name) {
+			case 'data-fullscreen':
+				this.fullscreen = Boolean(newval.toLowerCase());
```
Comment (KearseTrevor) [inline components/d2l-work-to-do/d2l-work-to-do.js:157]: Per conversation, needs to be changed to a different solve - current always evaluates to truthy

File: components/d2l-work-to-do/d2l-work-to-do.js [hunk 4]
```diff
@@ -139,6 +147,29 @@ class WorkToDoWidget extends EntityMixinLit(LocalizeMixin(LitElement)) {
...
+				break;
+			case 'data-upcoming-week-limit':
+				this._upcomingWeekLimit = parseInt(newval);
+				window.D2L.workToDoOptions.upcomingWeekLimit = this._upcomingWeekLimit;
+				break;
+			case 'data-overdue-week-limit':
+				this._overdueWeekLimit = parseInt(newval);
+				window.D2L.workToDoOptions.overdueWeekLimit = this._overdueWeekLimit;
+				break;
+		}
```
Comment (KearseTrevor) [inline components/d2l-work-to-do/d2l-work-to-do.js:168]: We may want to add protection against negative numbers

**Review by KearseTrevor — APPROVED** [review]

**Pull request merged**